sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -364,15 +366,28 @@ impl<'vm, 'state, 'a> ExecutionContext<'vm, 'state, 'a> {
         {
             return Err(CommandArgumentError::InvalidGasCoinUsage);
         }
-        // Immutable objects and shared objects cannot be taken by value
-        if matches!(
-            input_metadata_opt,
-            Some(InputObjectMetadata {
-                owner: Owner::Immutable | Owner::Shared { .. },
-                ..
-            })
-        ) {
-            return Err(CommandArgumentError::InvalidObjectByValue);
+        if protocol_config.shared_object_deletion() {
+            // Immutable objects cannot be taken by value
+            if matches!(
+                input_metadata_opt,
+                Some(InputObjectMetadata {
+                    owner: Owner::Immutable,
+                    ..
```

Comment:
We will need to continue to error _somehow_ when the `command_kind` is `TransferObjects`. It might be easiest here to match against `command_kind` when it is shared, and make sure we are only allowing the commands we want

Reply:
Ahh thank you for noticing this